service/controller/resource/machinedeploymentstatus/resource.go

```diff
@@ -82,8 +88,8 @@ func (r *Resource) ensure(ctx context.Context, obj interface{}) error {
 	}
 
 	{
-		cr.Status.Replicas = cc.Status.Worker[cr.Labels[label.MachineDeployment]].Nodes
-		cr.Status.ReadyReplicas = cc.Status.Worker[cr.Labels[label.MachineDeployment]].Ready
+		cr.Status.Replicas = workerCount[cr.Labels[label.MachineDeployment]].Nodes
+		cr.Status.ReadyReplicas = workerCount[cr.Labels[label.MachineDeployment]].Ready
```

Comment:
What is really cool with this is that the handlers for Machine Deployments and Control Planes are more or less the same. So we could see how we improve or even generate this in the future. Maybe there are patterns we can reuse more.